### PR TITLE
Revocation check (EXPOSUREAPP-12557)

### PR DESCRIPTION
**Pull request by mtwalli**
Review after #5058
## Changes
🪨 Cleanup old `Blocked` logic
🪨 Introduced `DccValidityMeasuresObserver` and `DccValidityMeasures` to have all state validation in one data flow
🪨 Refactored `DccStateChecker` which is main checker for all other checks and now provide every measure to the respective Checker
🪨 ofc implemented `Revoked` state checker

## Testing
- You would need a working API for that
<details>
  <summary> Alternatively </summary>
on `wru` You can provide the following hardcoded list below in `RevocationRepository` and scan 
 
  
```kotlin
        listOf(
                  CachedRevocationChunk(
                      coordinates = RevocationEntryCoordinates(
                          kid = "138291851ea0ddb4".decodeHex(),
                          type = RevocationHashType.SIGNATURE,
                          x = "26".decodeHex(),
                          y = "98".decodeHex()
                      ),
                      revocationChunk = RevocationChunk(
                          hashes = listOf(
                              "2698afff2cdbd017bc957b6fa1764ac7".decodeHex()
                          )
                      )
                  ),
                  CachedRevocationChunk(
                      coordinates = RevocationEntryCoordinates(
                          kid = "75d52ef3c0a6be96".decodeHex(),
                          type = RevocationHashType.UCI,
                          x = "54".decodeHex(),
                          y = "a9".decodeHex()
                      ),
                      revocationChunk = RevocationChunk(
                          hashes = listOf(
                              "54a90383b0e792489835caec8befd2a4".decodeHex()
                          )
                      )
                  ),
                  CachedRevocationChunk(
                      coordinates = RevocationEntryCoordinates(
                          kid = "9795a12910235933".decodeHex(),
                          type = RevocationHashType.COUNTRYCODEUCI,
                          x = "0c".decodeHex(),
                          y = "32".decodeHex()
                      ),
                      revocationChunk = RevocationChunk(
                          hashes = listOf(
                              "0c32d9ec22eb7167ea0e7a1eb0a21849".decodeHex()
                          )
                      )
                  ),
              )
```
</details>

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/statecheck/DccValidityMeasuresObserver.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        dccWalletInfoRepository.blockedQrCodeHashes,
+        revocationRepository.revocationList
+    ) { dscSignatureList, blockedQrCodeHashes, revocationList ->
+        DccValidityMeasures(
+            dscSignatureList = dscSignatureList,
+            blockedQrCodeHashes = blockedQrCodeHashes,
+            revocationList = revocationList
+        )
+    }.distinctUntilChanged().shareLatest(scope = appScope)
+
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/statecheck/DccValidityMeasuresObserver.kt:37]: ShareLatest uses a stateful flow internally which ist distingUntilChanged by design

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/revocation/calculation/RevocationCalculation.kt [hunk 2]
```diff
@@ -16,15 +35,22 @@ fun CwaCovidCertificate.calculateRevocationEntryForType(type: RevocationHashType
     }.also { Timber.tag(TAG).d("revocationEntry=%s", it) }
 }
 
-private fun CwaCovidCertificate.calculateRevocationEntryTypeUCI(): ByteString = uniqueCertificateIdentifier
+fun DccData<out DccV1.MetaData>.kidHash(): ByteString {
+    val kidHash = kid.decodeBase64()
+    Timber.tag(TAG).d("kidHash($kidHash)")
+    return kidHash ?: error("Bad KID!")
+}
```
Comment (BMItr) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/revocation/calculation/RevocationCalculation.kt:42]: What about:  return checkNotNull(kid) { "Bad KID" }

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/statecheck/DccValidityMeasuresObserver.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/statecheck/DccValidityMeasuresObserver.kt:36]: done

**Review by BMItr — APPROVED** [review]

**Review by jurajkusnier — APPROVED** [review]
LGTM

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/revocation/calculation/RevocationCalculation.kt (hunk 2 above)
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/revocation/calculation/RevocationCalculation.kt:42]: 🐰 
.. I meant kidHash
-> requireNotNull(kidHash)..
:-) sry

**Pull request merged**